pkg/register/component.go

```diff
@@ -197,73 +180,62 @@ func (c *component) RegisterUser(ctx context.Context, targetRealmName, customerR
-	authToken, err := c.generateAuthToken()
-
-	var userID string
+func (c *component) createUser(ctx context.Context, accessToken string, realmName string, user apiregister.UserRepresentation, groupNames []string, autoLoginToken keycloakb.TrustIDAuthToken) (string, string, error) {
 	var kcUser = user.ConvertToKeycloak()
-	kcUser.SetAttributeString(constants.AttrbTrustIDAuthToken, authToken.ToJSON())
 
-	if existingKcUser == nil {
-		userID, err = c.createKeycloakUser(ctx, accessToken, targetRealmName, &kcUser)
-		if err != nil {
-			return "", "", err
-		}
-	} else {
-		var realmConf = c.realmConfigurations[targetRealmName]
-		userID = *existingKcUser.ID
-		kcUser.ID = existingKcUser.ID
-		kcUser.Username = existingKcUser.Username
-		kcUser.Groups = &realmConf.endUserGroupIDs
-
-		err = c.keycloakClient.UpdateUser(accessToken, targetRealmName, userID, kcUser)
-		if err != nil {
-			c.logger.Warn(ctx, "msg", "Failed to update user through Keycloak API", "err", err.Error())
-			return "", "", err
-		}
+	// Set groups
+	groupIDs, err := c.convertGroupNamesToGroupIDs(accessToken, realmName, groupNames)
+	if err != nil {
+		c.logger.Error(ctx, "msg", "Failed to convert groupNames to groupIDs", "err", err.Error())
+		return "", "", err
 	}
+	kcUser.Groups = &groupIDs
 
-	// Store user in database
-	err = c.usersDBModule.StoreOrUpdateUserDetails(ctx, targetRealmName, dto.DBUser{
+	// Set authToken for auto login at the end of onboarding process
+	kcUser.SetAttributeString(constants.AttrbTrustIDAuthToken, autoLoginToken.ToJSON())
+
+	userID, username, err := c.createKeycloakUser(ctx, accessToken, realmName, kcUser)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Failed to update user through Keycloak API", "err", err.Error())
+		return "", "", err
+	}
+
+	// Store user details in database
+	err = c.usersDBModule.StoreOrUpdateUserDetails(ctx, realmName, dto.DBUser{
 		UserID:               &userID,
 		BirthLocation:        user.BirthLocation,
 		IDDocumentType:       user.IDDocumentType,
 		IDDocumentNumber:     user.IDDocumentNumber,
 		IDDocumentExpiration: user.IDDocumentExpiration,
 	})
+
 	if err != nil {
 		c.logger.Warn(ctx, "msg", "Can't store user details in database", "err", err.Error())
 		return "", "", err
 	}
 
-	// Send execute actions email
-	if err = c.sendExecuteActionsEmail(ctx, accessToken, targetRealmName, authToken, &kcUser, customerRealmName, userID, realmConf); err != nil {
-		return "", "", err
-	}
-
-	return userID, *kcUser.Username, nil
+	return userID, username, nil
 }
 
-func (c *component) createKeycloakUser(ctx context.Context, accessToken, targetRealmName string, kcUser *kc.UserRepresentation) (string, error) {
+func (c *component) createKeycloakUser(ctx context.Context, accessToken string, realmName string, kcUser kc.UserRepresentation) (string, string, error) {
```

Comment:
If you keep *kc.UserRepresentation, you can update kcUser with new ID and new username and just return a single error